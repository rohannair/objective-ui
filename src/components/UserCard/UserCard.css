@import '../../styles/globals';
@import '../../styles/mixins';
@import '../../styles/layout';

.UserCard {
  display: flex;
  width: 100%;
}

.header,
.body,
.footer {
  display: flex;
  flex-direction: column;
  padding: 20px;
  color: $secondary;
}

.header {
  align-items: center;
  border-bottom: 1px solid #f0f0f0;
  flex-direction: column;
}

.avatar {
  width: 64px;
  height: 64px;
  border-radius: 50%;
  overflow: hidden;
  margin: 0 0 10px;
}

.body {
  flex-direction: row;
  padding: 10px 0;
  border-bottom: 1px solid #f0f0f0;
  justify-content: baseline;
  align-items: baseline;
}

.missions,
.squads {
  padding: 10px;
  display: flex;
  flex-direction: column;
  flex: 1 0 auto;
}

.squads {
  border-right: 1px solid #f0f0f0;
}
