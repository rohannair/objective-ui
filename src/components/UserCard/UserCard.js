import React, { PropTypes } from 'react';
import styles from './UserCard.css';
import classNames from 'classnames/bind';

import Card from '../Card';
import Pill from '../Pill';

let cx = classNames.bind(styles);

const UserCard = ({ user }) => {
  const classname = cx({
    [styles.UserCard]: true
  });

  const userimg = user.img || '//placehold.it/250x250/eee?text=\?';
  const nameBox = user.firstName && user.lastName
  ? `${user.firstName } ${user.lastName }`
  : user.email;

  const titleBox = user.pending
  ? <Pill>Invited</Pill>
  : user.jobTitle || 'No title';

  const returnSquadPill = (squads) => {
    const count = squads && squads.length || 0;
    if (count === 0) return <Pill danger>No Squads</Pill>;
    return count > 1
    ? <Pill info>{`In ${count} squads`}</Pill>
    : <Pill info>{ squads[0].name }</Pill>;
  };

  const returnObjectivesPill = (objectives) => {
    const count = objectives && objectives.length || 0;
    if (count === 0) return <Pill danger>No OKRs</Pill>;
    return <Pill warning>{`${count} ${count === 1 ? 'OKR' : 'OKRs'}`}</Pill>;
  };


  return (
    <Card>
      <div className={ styles.header }>
        <img className={ styles.avatar} src={ userimg } />
        <h3>{ nameBox }</h3>
        <small>{ titleBox }</small>
      </div>

      <div className={ styles.body }>
        <p className={styles.squads}>{ returnSquadPill(user.squads) }</p>
        <p className={styles.missions}>{ returnObjectivesPill(user.objectives) }</p>
      </div>

    </Card>
  );
};

export default UserCard;
