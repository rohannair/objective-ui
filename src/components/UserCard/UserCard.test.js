import React from 'react'
import UserCard from './index.js'

describe('Test suite for UserCard component', () => {
  it('UserCard should exist', () => {
    let wrapper = shallow(<UserCard user={{}}/>)
    expect(wrapper).toMatchSnapshot()
  })
})
