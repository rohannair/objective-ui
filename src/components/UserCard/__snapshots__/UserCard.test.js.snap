exports[`Test suite for UserCard component UserCard should exist 1`] = `
<Component>
  <div>
    <img
      src="//placehold.it/250x250/eee?text=?" />
    <h3 />
    <small>
      No title
    </small>
  </div>
  <div>
    <p>
      <Pill
        danger={true}>
        No OKRs
      </Pill>
    </p>
  </div>
</Component>
`;
